draco/tests/test_hard.py

```diff
@@ -977,3 +977,400 @@ def test_view_scale_conflict():
+    entity(field,root,temperature).
+    entity(mark,root,1).
+    entity(encoding,1,2).
+    entity(encoding,1,3).
+
+    attribute((encoding,channel),2,x).
+    attribute((encoding,field),2,temperature).
+
+    attribute((encoding,channel),3,y).
+    attribute((encoding,field),3,temperature).
+    """
+        )
+        == ["same_field_x_and_y"]
+    )
+
+
+def test_count_twice():
+    b = hard.blocks["count_twice"]
+    assert isinstance(b, Block)
+
+    assert no_violations(
+        b.program
+        + """
+    entity(mark,root,1).
+    entity(encoding,1,2).
+    entity(encoding,1,3).
+
+    attribute((encoding,channel),2,x).
+    attribute((encoding,aggregate),2,count).
+
+    attribute((encoding,channel),3,y).
+    """
+    )
+
+    assert no_violations(
+        b.program
+        + """
+    entity(mark,root,1).
+    entity(encoding,1,3).
+    entity(encoding,2,4).
+
+    attribute((encoding,channel),3,x).
+    attribute((encoding,aggregate),3,count).
+
+    attribute((encoding,channel),4,y).
+    attribute((encoding,aggregate),4,count).
+    """
+    )
+
+    assert (
+        list_violations(
+            b.program
+            + """
+    entity(mark,root,1).
+    entity(encoding,1,2).
+    entity(encoding,1,3).
+
+    attribute((encoding,channel),2,x).
+    attribute((encoding,aggregate),2,count).
+
+    attribute((encoding,channel),3,y).
+    attribute((encoding,aggregate),3,count).
+    """
+        )
+        == ["count_twice"]
+    )
+
+    assert (
+        list_violations(
+            b.program
+            + """
+    entity(mark,root,1).
+    entity(encoding,1,2).
+    entity(encoding,1,3).
+
+    attribute((encoding,channel),2,x).
+    attribute((encoding,aggregate),2,count).
+
+    attribute((encoding,channel),3,size).
+    attribute((encoding,aggregate),3,count).
+    """
+        )
+        == ["count_twice"]
+    )
+
+
+def test_aggregate_not_all_continuous():
+    b = hard.blocks["aggregate_not_all_continuous"]
+    assert isinstance(b, Block)
+
+    assert no_violations(
+        b.program
+        + """
+    entity(mark,root,1).
+    entity(encoding,1,2).
+    entity(encoding,1,3).
+
+    attribute((encoding,channel),2,x).
+    attribute((encoding,aggregate),2,mean).
+
+    attribute((encoding,channel),3,y).
+    attribute((encoding,aggregate),3,mean).
+
+    entity(scale,root,4).
+    attribute((scale,type),4,linear).
+    attribute((scale,channel),4,x).
+    """
+    )
+
+    assert no_violations(
+        b.program
+        + """
+    entity(mark,root,1).
+    entity(encoding,1,2).
+    entity(encoding,1,3).
+
+    attribute((encoding,channel),2,x).
+    attribute((encoding,aggregate),2,mean).
+
+    attribute((encoding,channel),3,y).
+
+    entity(scale,root,4).
+    attribute((scale,type),4,linear).
+    attribute((scale,channel),4,x).
+
+    entity(scale,root,5).
+    attribute((scale,type),5,ordinal).
+    attribute((scale,channel),5,y).
+    """
+    )
+
+    assert (
+        list_violations(
```

Comment:
Like before, it would be great if you added comments to explain what the kind of violation in the example is. It would be great if you could send a pull request with comments to make it easier to understand the tests.